### PR TITLE
Decompiled ovl_Bg_Jya_Zurerukabe

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c [hunk 1]
```diff
@@ -21,23 +36,161 @@ const ActorInit Bg_Jya_Zurerukabe_InitVars = {
...
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+extern Gfx D_06012340[];
+extern UNK_PTR D_06012508;
+
+void func_8089B440(BgJyaZurerukabe* this, GlobalContext* globalCtx, void* arg2, DynaPolyMoveFlag flags) {
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c:67]:
```suggestion
void func_8089B440(BgJyaZurerukabe* this, GlobalContext* globalCtx, void* arg2, s32 flags) {
```
[cf this](https://github.com/zeldaret/oot/pull/400#discussion_r491705572)

File: src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c [hunk 2]
```diff
@@ -21,23 +36,161 @@ const ActorInit Bg_Jya_Zurerukabe_InitVars = {
...
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, localC);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_jya_zurerukabe.c", 194,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_8089B4C8(BgJyaZurerukabe* this, GlobalContext* globalCtx) {
+    Player* player = (Player*)globalCtx->actorCtx.actorList[ACTORTYPE_PLAYER].first;
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c:83]:
```suggestion
    Player* player = PLAYER;
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c (hunk 2 above)
Comment (ethteck) [reply src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c:83]: oh lol oops. Permuter copy-paste issue

File: src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c [hunk 3]
```diff
@@ -21,23 +36,160 @@ const ActorInit Bg_Jya_Zurerukabe_InitVars = {
...
+    Actor_ProcessInitChain(thisx, sInitChain);
+
+    for (i = 0; i < ARRAY_COUNT(D_8089B9F0); i++) {
+        if (fabsf(D_8089B9F0[i] - this->dyna.actor.initPosRot.pos.y) < 1.0f) {
+            this->unk_168 = i;
+            break;
+        }
+    }
+
+    if (D_8089B9F8 == &D_8089B9F0[i]) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c:128]:
```suggestion
    if (i == ARRAY_COUNT(D_8089B9F0)) {
```

Comment (ethteck) [reply src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c:128]: Yeah I agree it'd look a lot better, but I don't think it'll work..I'll try though

Comment (ethteck) [reply src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c]: ...nevermind! worked :)

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**